### ISSUE TITLE
containsColumn throws ArrayIndexOutOfBoundsException, in combination with selectFields

### ISSUE THREAD

**Issue by mickaeltr** [uniVocity/univocity-parsers#316]
Hello,

It seems that `containsColumn` throws an `ArrayIndexOutOfBoundsException`, when used in combination with `selectFields`. The following test fails because of that:

```java
import com.univocity.parsers.csv.CsvParser;
import com.univocity.parsers.csv.CsvParserSettings;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayInputStream;

class CsvParserTest {

  @Test
  void testContainColumns() {
    var settings = new CsvParserSettings();
    settings.setHeaderExtractionEnabled(true);
    settings.selectFields("AAA", "BBB", "CCC");
    var parser = new CsvParser(settings);
    var inputStream = new ByteArrayInputStream("AAA,BBB\na,b".getBytes());
    parser.beginParsing(inputStream);
    Assertions.assertFalse(parser.getRecordMetadata().containsColumn("CCC"));
  }

}
```

```
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 2

	at com.univocity.parsers.common.ColumnMap.indexOf(ColumnMap.java:69)
	at com.univocity.parsers.common.DefaultContext.indexOf(DefaultContext.java:82)
	at com.univocity.parsers.common.record.RecordMetaDataImpl.containsColumn(RecordMetaDataImpl.java:481)
```

Thanks

**Comment by jbax** [comment]
Thanks for reporting this bug. I've fixed it and released a 2.8.2-SNAPSHOT build which should work as expected.

Thank you for using our parsers!

**Comment by jbax** [comment]
No problem I'll try to this later today after I fix a few other issues
